### ISSUE TITLE
Docs: Add explanations about attributes of each Core Block in the documentation (and in the source code)

### ISSUE THREAD

**Issue by juanmaguitar** [WordPress/gutenberg#64057]
The only official documentation available for core blocks is located at [Block Editor Handbook > Reference Guides > Core Blocks Reference](https://developer.wordpress.org/block-editor/reference-guides/core-blocks/). This page lacks detailed explanations about each core block’s attributes and points the user to the source code for more information.

Providing extra explanations about the attributes of core blocks is essential for developers to fully understand and utilize these blocks effectively.

I propose including JSDoc comments in the Edit component of each block to explain the props.attributes object received by every core block’s edit component.

Besides providing extra info on these attributes directly in the source code, these descriptions could also be used (with some custom extra code) by [gen-block-lib-list.js](https://github.com/WordPress/gutenberg/blob/trunk/bin/api-docs/gen-block-lib-list.js) to generate more extended info for each block at [Block Editor Handbook > Reference Guides > Core Blocks Reference](https://href.li/?https://developer.wordpress.org/block-editor/reference-guides/core-blocks/)

- [ ] [archives](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/archives)
- [ ] [audio](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/audio)
- [ ] [avatar](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/avatar)
- [ ] [block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/block)
- [ ] [block-keyboard-shortcuts](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/block-keyboard-shortcuts)
- [ ] [button](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/button)
- [ ] [buttons](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/buttons)
- [ ] [calendar](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/calendar)
- [ ] [categories](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/categories)
- [ ] [classic.scss](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/classic.scss)
- [ ] [code](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/code)
- [ ] [column](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/column)
- [ ] [columns](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/columns)
- [ ] [comment-author-avatar](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comment-author-avatar)
- [ ] [comment-author-name](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comment-author-name)
- [ ] [comment-content](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comment-content)
- [ ] [comment-date](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comment-date)
- [ ] [comment-edit-link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comment-edit-link)
- [ ] [comment-reply-link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comment-reply-link)
- [ ] [comment-template](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comment-template)
- [ ] [comments](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comments)
- [ ] [comments-pagination](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comments-pagination)
- [ ] [comments-pagination-next](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comments-pagination-next)
- [ ] [comments-pagination-numbers](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comments-pagination-numbers)
- [ ] [comments-pagination-previous](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comments-pagination-previous)
- [ ] [comments-title](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/comments-title)
- [ ] [common.scss](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/common.scss)
- [ ] [cover](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/cover)
- [ ] [details](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/details)
- [ ] [editor-elements.scss](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/editor-elements.scss)
- [ ] [editor.scss](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/editor.scss)
- [ ] [elements.scss](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/elements.scss)
- [ ] [embed](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/embed)
- [ ] [file](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/file)
- [ ] [footnotes](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/footnotes)
- [ ] [form](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/form)
- [ ] [form-input](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/form-input)
- [ ] [form-submission-notification](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/form-submission-notification)
- [ ] [form-submit-button](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/form-submit-button)
- [ ] [freeform](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/freeform)
- [ ] [gallery](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/gallery)
- [ ] [group](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/group)
- [ ] [heading](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/heading)
- [ ] [home-link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/home-link)
- [ ] [html](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/html)
- [ ] [image](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/image)
- [ ] [latest-comments](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/latest-comments)
- [ ] [latest-posts](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/latest-posts)
- [ ] [list](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/list)
- [ ] [list-item](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/list-item)
- [ ] [lock-unlock.js](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/lock-unlock.js)
- [ ] [loginout](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/loginout)
- [ ] [media-text](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/media-text)
- [ ] [missing](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/missing)
- [ ] [more](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/more)
- [ ] [navigation](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/navigation)
- [ ] [navigation-link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/navigation-link)
- [ ] [navigation-submenu](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/navigation-submenu)
- [ ] [nextpage](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/nextpage)
- [ ] [page-list](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/page-list)
- [ ] [page-list-item](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/page-list-item)
- [ ] [paragraph](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/paragraph)
- [ ] [pattern](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/pattern)
- [ ] [post-author](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-author)
- [ ] [post-author-biography](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-author-biography)
- [ ] [post-author-name](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-author-name)
- [ ] [post-comment](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comment)
- [ ] [post-comments-count](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments-count)
- [ ] [post-comments-form](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments-form)
- [ ] [post-comments-link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-comments-link)
- [ ] [post-content](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-content)
- [ ] [post-date](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-date)
- [ ] [post-excerpt](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-excerpt)
- [ ] [post-featured-image](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-featured-image)
- [ ] [post-navigation-link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-navigation-link)
- [ ] [post-template](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-template)
- [ ] [post-terms](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-terms)
- [ ] [post-time-to-read](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-time-to-read)
- [ ] [post-title](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/post-title)
- [ ] [preformatted](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/preformatted)
- [ ] [pullquote](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/pullquote)
- [ ] [query](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query)
- [ ] [query-no-results](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-no-results)
- [ ] [query-pagination](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination)
- [ ] [query-pagination-next](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination-next)
- [ ] [query-pagination-numbers](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination-numbers)
- [ ] [query-pagination-previous](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-pagination-previous)
- [ ] [query-title](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/query-title)
- [ ] [quote](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/quote)
- [ ] [read-more](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/read-more)
- [ ] [rss](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/rss)
- [ ] [search](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/search)
- [ ] [separator](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/separator)
- [ ] [shortcode](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/shortcode)
- [ ] [site-logo](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/site-logo)
- [ ] [site-tagline](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/site-tagline)
- [ ] [site-title](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/site-title)
- [ ] [social-link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/social-link)
- [ ] [social-links](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/social-links)
- [ ] [spacer](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/spacer)
- [ ] [table](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/table)
- [ ] [table-of-contents](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/table-of-contents)
- [ ] [tag-cloud](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/tag-cloud)
- [ ] [template-part](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/template-part)
- [ ] [term-description](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/term-description)
- [ ] [text-columns](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/text-columns)
- [ ] [verse](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/verse)
- [ ] [video](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/video)

**Comment by vipul0425** [comment]
Hi @juanmaguitar,

I’ve created an initial PR for this issue (https://github.com/WordPress/gutenberg/pull/68508). It includes the additional code changes to the `gen-block-lib-list.js` script and adds the documentation comment for the `archives` block.
For the JSDoc comment, I referred to the guidance shared here: https://github.com/WordPress/gutenberg/issues/22891#issuecomment-2543030487, as mentioned by @t-hamano.

Looking forward to your feedback.
Thanks 🙇

**Comment by stokesman** [comment]
I agree with the idea here. My initial thought on the proposal of adding JSDoc comments to the Edit components is that it introduces duplication and therefore seems less than ideal for maintenance.

Has adding this directly in block json been considered? There each attribute already supports some relevant metadata (`type` and `default`). What about adding a description for attributes? I can think of two ways that might be done. Either adding it as a property or supporting JSON5 and adding it as a comment.

Adding a `description` property to attributes seems the simpler idea. However, ideally their data wouldn’t contribute to any bundle size increases and for that it seems a custom JSON loader for Webpack that could strip such properties would be wanted. Additionally—though likely of lesser concern—the generation of `blocks-json.php` in core might also strip the properties.

Supporting JSON5 would probably be more work yet would seemingly have a broader appeal. Adding attribute descriptions in the way of comments should avoid the concern of bundle size increases. However, that might be less simple with regards to generating docs.

**Comment by Mamaduka** [comment]
Putting this proposal on hold until the following items are considered/resolved:

* Standardized way of documenting block attributes. I agree with @stokesman that JSDoc comments might not be the best option.
* A parser that can include these details in documentation. There's already one parsing block attribute name from the `block.json` file; it might be easier to reuse that.

tl;dr, we need to have a good plan before moving forward and updating dozens of files.

cc @vipul0425, @Anuj-Rathore24, @dhruvikpatel18

**Comment by t-hamano** [comment]
If we wanted to use JSON5 or JSONC to be able to insert comments, we'd probably have to convert all `block.json` to `block.json5` or `block.jsonc`, and the `@wordpress/scripts` would have to support the new extensions, right?

On the other hand, because `additionalProperties` is not defined in the `attributes` field, any field can be added, so it would be possible to add a field such as `description`. However, ideally additional fields should not be allowed.

**Comment by Mamaduka** [comment]
IIRC, JSON5 or JSONC aren't supported by default in PHP, so I doubt the format change.

P.S. There was a similar conversation for the `theme.json` format change, and I remember that was one of the reasons.
